src/lib/i18n.ts

```diff
@@ -1,47 +1,98 @@
 import i18n from "i18next";
 import Backend from "i18next-http-backend";
+import { useMemo } from "react";
 import {
-	type UseTranslationResponse,
 	initReactI18next,
 	useTranslation,
 } from "react-i18next";
+import { z } from "zod";
+import { zodI18nMap } from "zod-i18n-map";
+import {
+	default as ZodEnUs,
+	default as ZodNlNl,
+} from "zod-i18n-map/locales/nl/zod.json";
+
+/** Dynamic import of all json files in i18n,
+ * parse their filenames and use them as locales
+ * and load their url during build so they're statically
+ * built to prevent caching as static assets.
+ */
+const supportedLanguages = Object.fromEntries(
+	await Promise.all(
+		Object.entries(
+			await import.meta.glob("../i18n/*.json", {
+				query: "?url",
+				import: "default",
+			}),
+		).map(async ([path, resolveFilename]) => [
+			path
```

Comment:
Maybe add a comment here that we are extracting the language key `nl-NL` part. I had to log it myself to understand.